File: JZOffer/python/05.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # 返回从尾部到头部的列表值序列，例如[1,2,3]
    def printListFromTailToHead(self, listNode):
        if(not listNode):
            return []
        else:
            result = []
            while listNode:
                result.append(listNode.val)
                listNode = listNode.next
            return result.reverse()

File: JZOffer/js/05.js
/*function ListNode(x){
    this.val = x;
    this.next = null;
}*/
function printListFromTailToHead(head)
{
    if(!head){
        return {};
    }
    var arr = [];
    while(head){
        arr.push(head.val);
        head = head.next;
    }
    if(arr){
        return arr.reverse();
    }
}
